calculator/src/extension.cpp

```diff
@@ -86,8 +109,24 @@ QWidget *Calculator::Extension::widget(QWidget *parent) {
 /** ***************************************************************************/
 void Calculator::Extension::handleQuery(Core::Query * query) const {
 
+    bool hexParse = settings().value(CFG_HEXP, CFG_HEXP_DEF).toBool();
+
+    if(hexParse)
+    {
+        auto hexPrefix = QString("0x");
+        bool isHex = query->string().contains(hexPrefix);
+        hexParse = isHex;
+    }
+
     try {
-        d->parser->SetExpr(query->string().toStdString());
+        if(hexParse)
+        {
+            d->iparser->SetExpr(query->string().toStdString());
+        }
+        else
+        {
+            d->parser->SetExpr(query->string().toStdString());
+        }
```

Comment:
this block is weird

```cpp
if(d->iparser && query->string().contains("0x")) 
    d->iparser->SetExpr(query->string().toStdString());
else 
    d->parser->SetExpr(query->string().toStdString());
```

Reply:
I'd actually keep the curlies, but have them on the same line as if/else. That way it looks nicer, still allows editors to highlight the matching brace, *and* any future additions to the code inside them won't litter diffs just to add `{}` around it :wink: 

```cpp
if (d->iparser && query->string().contains("0x")) {
    // ...
} else {
    // ...
}
```